File: itest/containers/Containerfile.ts
# Copyright (c) 2026 Data In Motion and others.
# SPDX-License-Identifier: EPL-2.0
#
# DDSR TypeScript workspace image. Build with ddsr-ts-client/ as
# context (its .dockerignore keeps node_modules/dist out):
#
#   podman build -f itest/containers/Containerfile.ts -t ddsr/ts ddsr-ts-client/
#
# Run any example/probe by overriding the command:
#   podman run --network=host -e BROKER_URL=... ddsr/ts provider.ts
#   podman run --network=host -e BROKER_URL=... ddsr/ts harness-probe.ts
FROM docker.io/library/node:24-alpine
RUN corepack enable pnpm
WORKDIR /app
COPY . .
RUN corepack pnpm install --frozen-lockfile && corepack pnpm -r build
WORKDIR /app/examples/payment
# node directly (no pnpm wrapper): SIGTERM must reach the process so
# the FR-P3 shutdown hooks run.
ENTRYPOINT ["node", "--import", "tsx"]
CMD ["provider.ts"]

File: itest/containers/Containerfile.java
# Copyright (c) 2026 Data In Motion and others.
# SPDX-License-Identifier: EPL-2.0
#
# Generic runner for an exported bnd executable jar. Build with the
# module's generated/distributions/executable/ directory as context:
#
#   podman build -f itest/containers/Containerfile.java \
#     --build-arg JAR=broker.jar -t ddsr/broker \
#     org.eclipse.fennec.services.broker.rest/generated/distributions/executable/
FROM docker.io/library/eclipse-temurin:21-jre
ARG JAR
COPY ${JAR} /app/app.jar
WORKDIR /work
# --nointeractive: without a TTY the gogo shell would stop the framework.
ENTRYPOINT ["java", "-Dgosh.args=--nointeractive", "-jar", "/app/app.jar"]
